server/gitrest/packages/gitrest-base/src/utils/gitWholeSummaryManager.ts

```diff
@@ -668,10 +668,48 @@ export class GitWholeSummaryManager {
 			}
 		}
 
-		const inMemorySummaryFullGitTree = await this.writeSummaryTreeCore(
-			wholeSummaryTreeEntries,
-			inMemoryRepoManager,
-		);
+		const writeSummaryIntoMemory = async () =>
+			this.writeSummaryTreeCore(wholeSummaryTreeEntries, inMemoryRepoManager);
+		const inMemorySummaryFullGitTree = await writeSummaryIntoMemory().catch(async (error) => {
+			if (
+				error?.caller === "git.walk" &&
+				error.code === "NotFoundError" &&
+				typeof error.data?.what === "string"
+			) {
+				// This is caused by the previous channel summary tree being missing.
+				// Fetch the missing tree, write it into the in-memory storage, then retry.
+				const missingTreeSha = error.data.what;
+				const missingTree = await this.repoManager.getTree(
+					missingTreeSha,
+					true /* recursive */,
```

Comment:
Yes it will read from storage, but overall it will not be a greater amount than `enableLowIoWrite !== true` which is the whole point of this flag. This will have a higher request cost the first time converting from high-io to low-io, because the Channel tree will be stored as individual blobs. After conversion though, it will be a lower cost because the Channel tree will always be stored as 1 single blob, so then this line will only be 3 reads: 1 tree with 2 entries, and 1 blob.

Comparing that to even the simplest Channel tree with only a quorum change, which has 6 tree entries (3 tree, 3 blob), this will be less, and also will reduce write cost when writing a summary. For a slightly more complex Channel Tree generated by my _very simple_ chat application, I can see Channel Trees with `"treeNodeCount":13, "blobNodeCount":28, "handleNodeCount":51`, so reducing that to 3 reads/writes from 92 will be largely impactful.